app/src/main/java/com/arctouch/codechallenge/movies_list/presentation/MoviesListPresenter.java

```diff
@@ -0,0 +1,227 @@
+package com.arctouch.codechallenge.movies_list.presentation;
+
+import android.content.Context;
+import android.util.Log;
+
+import com.arctouch.codechallenge.BasePresenter;
+import com.arctouch.codechallenge.R;
+import com.arctouch.codechallenge.api.TmdbApi;
+import com.arctouch.codechallenge.data.Cache;
+import com.arctouch.codechallenge.di.InjectionTmdbApi;
+import com.arctouch.codechallenge.model.Genre;
+import com.arctouch.codechallenge.model.GenreResponse;
+import com.arctouch.codechallenge.model.Movie;
+import com.arctouch.codechallenge.model.UpcomingMoviesResponse;
+
+import java.io.IOException;
+import java.net.SocketTimeoutException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import io.reactivex.Observer;
+import io.reactivex.Scheduler;
+import io.reactivex.disposables.Disposable;
+import retrofit2.Response;
+
+public class MoviesListPresenter extends BasePresenter<MoviesListContract.View> implements MoviesListContract.Presenter {
+
+    private Context context;
+    private Scheduler scheduler;
+    private Scheduler observer;
+
+    public MoviesListPresenter(Context context, MoviesListContract.View view, Scheduler scheduler, Scheduler observer) {
+        super(view);
+        this.context = context;
+        this.scheduler = scheduler;
+        this.observer = observer;
+    }
+
+    @Override
+    public void getGenres(boolean hideProgress) {
+        view.showProgress();
+
+        InjectionTmdbApi.inject()
+                .genres(TmdbApi.API_KEY, TmdbApi.DEFAULT_LANGUAGE)
+                .subscribeOn(scheduler)
+                .observeOn(observer)
+                .subscribe(new Observer<Response<GenreResponse>>() {
```

Comment:
What would happen if the user rotated the device after this call but before the response returns?